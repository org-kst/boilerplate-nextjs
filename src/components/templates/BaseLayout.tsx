import React from 'react';

export const BaseLayout = (props: { children: React.ReactNode }) => {
    return (
        <div>
            <header></header>
            <main>{props.children}</main>
            <footer></footer>
        </div>
    );
};
