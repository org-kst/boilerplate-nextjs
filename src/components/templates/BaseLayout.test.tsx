import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { BaseLayout } from './BaseLayout';

describe('BaseLayout', () => {
    describe('render', () => {
        it('should have header tag.', () => {
            const { container } = render(<BaseLayout>test</BaseLayout>);

            const header = container.querySelector('header');
            expect(header).toBeInTheDocument();
        });
    });
    it('should match snapshot.', () => {
        const { container } = render(<BaseLayout>test</BaseLayout>);

        expect(container).matchSnapshot();
    });
});
