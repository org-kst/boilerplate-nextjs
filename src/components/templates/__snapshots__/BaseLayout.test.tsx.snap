// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseLayout > should match snapshot. 1`] = `
<div>
  <div>
    <header />
    <main>
      test
    </main>
    <footer />
  </div>
</div>
`;
